### DISCUSSION TITLE
Drop Python 3.7, Python 3.8, (3.9?, 3.10??) support?

### DISCUSSION THREAD

**Discussion by jbeckerwsi** [FreeOpcUa/opcua-asyncio#1713] [General]
With Python 3.7 being EOL since 27 Jun 2023 and 3.8 loosing security support in 4 weeks on 31 Oct 2024, **what are your thoughts** on dropping support for these version, or at least 3.7?

Support in this package for 3.6 was dropped 4 years ago in https://github.com/FreeOpcUa/opcua-asyncio/commit/b5c770800cfdccee445f5653e78b7f3ff3a7a73c

Since this package is targeted at industrial control systems, I can see how supporting legacy versions might be desirable, but I'd still like to put in a vote to drop support for as many old python versions, up to 3.10 (leaving >=3.11) as possible.

**Comment by AndreasHeine** [comment]
software ages like milk not like wine ^^

i prefer dropping!

**Reply by AndreasHeine** [reply]
3.6 to 3.7 was a big change asyncio related if i am not mistaken...

**Reply by jbeckerwsi** [reply]
@AndreasHeine are you thinking to just go for it or would you like to wait for some more user voices?
And if so, which versions would you drop? I can come with a PR in the next couple of days if that would be helpful.

**Reply by AndreasHeine** [reply]
1. i mean EoL is EoL... 
2. what has changed and are we using the new features of Python!?
3. Older Python version might work still the question is do we hold up the promise its working on legacy versions

**Reply by jbeckerwsi** [reply]
I am quite fond of the improvements made to the type system in recent versions. It would make adding more type hints to the projects easier and more readable. 3.10 for example adds the "A | B" syntax for type unions.
Another nice addition in 3.10 was structural pattern matching, likely there are good use cases here. I personally found it rather useful when handling data coming in through any kind of communication channel.
